### PR TITLE
WIP: CMSP-9: Provides the multisite variable for the repo automatically.

### PR DESCRIPTION
**Pull request by rwagner00**
### Overview
This pull request:

| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no 
| Has tests?    | yes/no
| BC breaks?    | no     
| Deprecations? | yes/no 

### Summary
Short overview of what changed.

### Description
Any additional information.

### CODE REVIEW

**Review by greg-1-anderson — CHANGES_REQUESTED** [review]
I have reservations.

- **CAUTIONARY NOTE:** You're rewriting tags of old versions of wordpress-network. We've done things like this in the past; have you discussed as a team and decided that it's what you want here? The alternative would be to only make the change to new tags.
- You're deleting old versions of the wordpress-network tags on every update, though, which means that you're rewriting history for all tags in the project on every release. You shouldn't do this. **REQUIRED CHANGE:** If you need to rewrite the tags, delete them manually when you're ready, and have the tool continue to only provide missing tags.
- I don't like the upstream-specific method names. I'd prefer to see a filter-based mechanism for something like this, similar to existing update-tool functionality. **VETOABLE** by @CdrMarks or @scottbuscemi 
- You're using a patch file to update the wp-config-panthoen.php file, rather than procedural code. Patch files are fragile and may start producing merge conflicts at inopportune times. **VETOABLE** by @CdrMarks or @scottbuscemi 

**RECOMMENDED ALTERNATIVE** Have you considered providing this functionality some other way? For example, could you inspect the ygg variables, and only define WP_ALLOW_MULTISITE to `true` if the framework is wordpress-network? If you did that, then the code in the wp-config-panthon.php could be the same for both upstreams, and the update tool would not need to change at all.

**Comment by rwagner00** [comment]
It won't rewrite old tags, those don't get processed at all, only the new ones are changed.

The logic of the thing doesn't delete the tags, per se. The existing code simply checks out the tag x.y.z from the WP repo into the WP-Network working repo and then pushes that tag up immediately. Our objective is for the x.y.z tag to include the new change, so the tag needs to be regenerated, which requires deleting the tag, making the change and then recreating before pushing it up.

I'm not sure what you mean here, can you provide some additional context?

Valid point, I can switch the patch file for a procedural code approach. 

I hadn't really considered that alternative approach mostly because the ticket explicitly called for the work to be done by changing the update tool. I'd be happy to discuss that though!

**Comment by greg-1-anderson** [comment]
Do not merge; `wordpress-network` is an upstream. If you rewrite history, you will break dashboard updates for all WordPress network sites. See slack discussion: https://pantheon.slack.com/archives/C024NA0Q1TJ/p1676676556259349?thread_ts=1676671780.729339&cid=C024NA0Q1TJ

File: src/Cli/ProjectCommands.php [hunk 1]
```diff
@@ -194,6 +210,43 @@ public function projectDerivativePull($remote, $options = ['as' => 'default', 'p
...
+     * Performs minor updates for the WordPress Site Network repo.
+     *
+     * @param WorkingCopy $project_working_copy
+     *   The remote repo being worked on.
+     *
+     * @return void
+     */
+    protected function wordpressNetworkPostTagUpdates(WorkingCopy $project_working_copy, $version)
+    {
+        $project_working_copy->tagDelete($version);
```
Comment (greg-1-anderson) [inline src/Cli/ProjectCommands.php:240]: If this code only runs on **new** tags, then this line should not be needed at all. This PR is much less alarming without this line, and could be re-opened if that was your preference. I'd still strongly prefer to see my recommended alternative implementation (in the WordPress upstream) though.

**Pull request closed without merging**